### ISSUE TITLE
Doom 3 (9050)

### ISSUE THREAD

**Issue by adrianlshaw** [ValveSoftware/Proton#713]
This is the original version of the game, not the BFG edition (discussed in #569 ). Videos are working fine. Gameplay too.

Specs: Proton 3.7-3, Ubuntu 18.04, NVIDIA Corporation G94GL [Quadro FX 1800] (rev a1)
Full specs: https://gist.github.com/adrianlshaw/13b21eae4d9ad3c6c79474396110feed

**Comment by gort818** [comment]
I wish we could use the Linux binary instead of running it through proton.

**Comment by kode54** [comment]
Well, technically, you could also use the newer BFG edition that combines both of the campaigns into a single package, and that definitely requires Proton, because it was never ported from Windows.

**Comment by andrew-mcmahon** [comment]
Whilst it's nice to see this working through Wine/Proton; I'd imagine most people will be better off compiling:

- RBDoom-3-BFG: https://github.com/RobertBeckebans/RBDOOM-3-BFG (for BFG edition)
- dhewm3: https://github.com/dhewm/dhewm3 (for vanilla doom3)

There seems to be packages available in the repos for Ubuntu, Arch, etc.
Perhaps Valve could find a seamless way to integrate projects like these into Steam somehow. 
Something better than the "add a non-steam game" route - a workshop mod?

**Comment by xDShot** [comment]
@andrew-mcmahon RBDoom-3-BFG is unstable and misses some important features like achievements and online multiplayer. dhewm3 is also unstable, contains bugs not presented in vanilla and incompatible with most existing mods. Unless these two projects find workaround and can deal with publishers (in which I doubt), I wouldn't want them to appear on Steam.

**Comment by xDShot** [comment]
Found glitches:
Glass windows are fully opaque:
![01](https://user-images.githubusercontent.com/4261659/44832188-7bca5500-ac32-11e8-8370-6137f1ad642d.jpg)
![02](https://user-images.githubusercontent.com/4261659/44832189-7bca5500-ac32-11e8-8769-55a4ee5d9e3c.jpg)
![03](https://user-images.githubusercontent.com/4261659/44832191-7c62eb80-ac32-11e8-86b4-6123c5cab434.jpg)
Glass + messed up UI image:
![04](https://user-images.githubusercontent.com/4261659/44832214-98668d00-ac32-11e8-8b9c-6ac362663e1c.jpg)

**Comment by kode54** [comment]
@xDShot Post your system info in a gist, please.

I'm attaching (resized, I forgot that this game is hardcoded to 4:3 aspect) screen shots to show things work on my system.

![20180829233816_aspect](https://user-images.githubusercontent.com/796316/44836294-6847cb80-abeb-11e8-971c-2377859f205f.jpg)
![20180829234005_aspect](https://user-images.githubusercontent.com/796316/44836309-71389d00-abeb-11e8-9ede-c3104f6a4d27.jpg)
![20180829234021_aspect](https://user-images.githubusercontent.com/796316/44836316-7564ba80-abeb-11e8-84da-91c502818d06.jpg)
![20180829234206_aspect](https://user-images.githubusercontent.com/796316/44836320-7990d800-abeb-11e8-9b40-afa0edf7773d.jpg)
![20180829234208_aspect](https://user-images.githubusercontent.com/796316/44836325-7c8bc880-abeb-11e8-98eb-129fc4865caa.jpg)
![20180829234536_aspect](https://user-images.githubusercontent.com/796316/44836329-7eee2280-abeb-11e8-8874-702d928739df.jpg)
![20180829234939_aspect](https://user-images.githubusercontent.com/796316/44836333-81507c80-abeb-11e8-95bf-2f6207845f79.jpg)
![20180830001706_aspect](https://user-images.githubusercontent.com/796316/44836337-831a4000-abeb-11e8-8240-3df027e1363c.jpg)

System:
https://gist.github.com/kode54/3ff952f39b407d6f806a6779838f5335

Using the RX480 listed there.

**Comment by xDShot** [comment]
https://gist.github.com/xDShot/072b363fb24811836d146ce3242887ab

**Comment by DanielGibson** [comment]
> dhewm3 is also unstable, contains bugs not presented in vanilla

Is it? Could you file bugreports for those bugs and instabilities at https://github.com/dhewm/dhewm3/issues ?

**Comment by andrew-mcmahon** [comment]
I don't suffer those graphical glitches either so it must be specific to Intel integrated gfx.
[Steam system info](https://gist.github.com/andrew-mcmahon/7dedc63216715cbcf8225da226a2da5e)
I would've thought that [online multiplayer is completely dead](https://steamdb.info/app/9050/graphs/) anyway? (at least according to steamDB)

What I will say is that enabling HDR in **RBDoom3-BFG** seems to be the cause of some rendering issues.
Drop down a console (¬) and turn that off with **r_useHDR 0**
Soft shadows are also _very demanding_; the basic stencil shadows are OK.

No real issues with **dhewm3** either. 
I've noticed the older engine stutters a bit in places and the GUI + videos are 4:3 as they were originally.
Even if the GUI can't be fixed easily; non-stretched/letterboxed videos would be a nice addition.

RBDoom3-BFG screenshots with GALLIUM HUD @ 1080p
![screenshot_2018-08-30_15-29-36](https://user-images.githubusercontent.com/42167250/44861442-712eb080-ac70-11e8-9644-082e92465834.png)
![screenshot_2018-08-30_15-30-59](https://user-images.githubusercontent.com/42167250/44861453-75f36480-ac70-11e8-9fee-062124f3ad71.png)

dhewm3 screenshots with GALLIUM HUD @ 1080p
![screenshot_2018-08-30_16-23-43](https://user-images.githubusercontent.com/42167250/44861831-498c1800-ac71-11e8-8f8d-1994d14d4d6c.png)
![screenshot_2018-08-30_16-24-06](https://user-images.githubusercontent.com/42167250/44861834-4abd4500-ac71-11e8-9db3-41187851305f.png)

A few more comparison pics [here](https://imgur.com/a/UiKSVfw) and [here](https://imgur.com/a/NfDVN1y).
Personally I love having the choice of which edition of Doom 3 to use.

**Comment by flibitijibibo** [comment]
@kisak-valve Requesting an XAudio2 label for this game...

**Comment by DanielGibson** [comment]
Doom3 from 2004 uses XAudio2? (Very unlikely, unless they updated the soundcode for the Steam release)
Or did you mean Doom3 BFG, which is discussed at https://github.com/ValveSoftware/Proton/issues/569 ?

**Comment by flibitijibibo** [comment]
Oh, shoot, i clicked the wrong one. Dan is right, I meant the other one, original Doom3 uses OpenAL!

**Comment by legluondunet** [comment]
@andrew-mcmahon  even if there exist some modern port, this games should simply work with Proton, like the others WIndows games on Linux, that the subject here. Advanced users could always try others ports. I just tested Doom3 with Proton 3.16-3 and it works very well. You click, you play! @kisak-valve It could be whitelisted.
https://gist.github.com/legluondunet/0f0a56fb4e9cb743f111385e554fce69

**Comment by oblitum** [comment]
It doesn't launch at all on Intel 7700K with just Intel HD graphics.

**Comment by oblitum** [comment]
Game is working fine now (Proton 3.16-6 Beta).

**Comment by kmaterka** [comment]
Doom 3: BFG Edition does not start with Proton 4.2-2. Works correctly with 3.16-8 Beta. Can you verify if it also affects you?

**Comment by kisak-valve** [comment]
Hello @kmaterka, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options, try the game with Proton 4.2-2, and drag and drop the generated $HOME/steam-$APPID.log into the comment box.

**Comment by kmaterka** [comment]
@kisak-valve Sure, log attached:
[steam-208200.zip](https://github.com/ValveSoftware/Proton/files/3075293/steam-208200.zip)

**Comment by kode54** [comment]
@kmaterka Here’s a shot in the dark: Try removing OpenVR?

**Comment by kmaterka** [comment]
@kode54 Good shot! After removing these two:
./steamapps/compatdata/208200/pfx/drive_c/windows/syswow64/openvr_api_dxvk.dll
./steamapps/compatdata/208200/pfx/drive_c/windows/system32/openvr_api_dxvk.dll
game started.

**Comment by kode54** [comment]
Although, specifically, it appears it was the input code crashing somewhere related to libfreespace, but that only appears to get loaded if the OpenVR framework is present. It would be beneficial to those attempting to run the VR integration on Linux to fix this, but until then, it would also be nice if the game’s current VR libraries didn’t crash in Wine even without OpenVR installed in Steam, unless those libraries came extra because of something you did.

**Comment by kisak-valve** [comment]
Doom 3 (original not BFG) has a regression that makes it impossible to control (mouselook) since Proton 9

Issue transferred from https://github.com/ValveSoftware/Proton/issues/9007.
@Deus-nsf posted on 2025-08-26T19:12:10:

I must clarify that I'm on the SteamDeck (even though that shouldn't really change much).

Reverting to Proton 8 fixes the issue, but more recent Proton are useful for some features, so it's a bit of a shame to have to resort to a downgrade. The issue seems to affect the way mouse control works for the game camera, it can also be reproduced on Proton 10 beta, Proton Exp, and Proton Hotfix, as well as many community branches that are based on 9 and up.

I have been hesitating on reporting this for a while as I thought it might be an issue on my end but it doesn't seem so, also more recent Proton are necessary for LSFG-VK to have PROTON_USE_WOW64=1 otherwise 32 bit apps won't work. And since OG Doom 3 is locked at 62.5 FPS, the newer features are appreciated.

**Comment by alasky17** [comment]
@Deus-nsf Could you please try stock 9.0-4 and see if this build has the mouselook issue that you are describing?  I *think* I see the same thing as you, but I cannot reproduce it on 9.0-4 or experimental-9-bleeding-edge (which is a beta option available in the Proton - Experimental beta dropdown list called "bleeding-edge-9.0").  It would be really helpful if you could test 9.0-4 and the 9.0-bleeding-edge builds to see if the regression you are seeing is between Proton 8 -> 9 or 9 -> 10 or 9 -> 9.0-bleeding-edge.

I am using the Keyboard (WASD) and Mouse layout on the deck, and what I'm seeing is that the camera is sometimes jumping (particularly in the vertical direction) when I spin in a circle with the trackpad to control the camera and act as the "mouse".  Is this similar to the "mouselook" issue you are seeing?  If not, could you give some more description about the problem you are seeing and the controller settings you are using?

**Comment by Deus-nsf** [comment]
Yes! It is this exact issue, as far as I can remember it does it on stock 9.0-4, let me double check and get back to you (will edit this comment).

EDIT: it looks like stock 9.0-4 doesn't have this issue, but an other one! the camera movements are a bit juddery, but when you only strafe and don't move the camera it's fine. So the issues starts with 9.0-4 but not the one I initially described!

I'm playing with the original engine, and I've been trying with the engine's config variable "m_smooth=1" and "m_smooth=0" but no difference.

**Comment by alasky17** [comment]
@Deus-nsf Thank you for the confirmation! I took another look and I'm not noticing any difference between 8.0 and 9.0 ... I saw you mention multiple times that you are using the original engine ... is this an option I need to toggle somewhere?  I'm just launching the game and using the default options - the appid is 9050, so I believe we are testing the same version of the game in general. Separately, the 9 -> 10 issue is still definitely a regression that I'll look into deeper :)

**Comment by alasky17** [comment]
@Deus-nsf Also - could you please try the experimental "bleeding-edge" branch? You will need to manually select "bleeding-edge" as the beta option for Proton - Experimental.  A different regression was just fixed this week, and the behavior seems better to me with bleeding-edge than default experimental.  Are you seeing an improvement as well?  Is it still worse than 9?

**Comment by Deus-nsf** [comment]
Thanks a lot for taking a look at this again, when I mention "original engine", it just means that it's the vanilla default engine that ships with the Steam release, I mentionned it because there's several quality source ports for Doom 3, but I'm using none of them for the deck, just the regular .ini tweaks to put the game in 16/9, that sort of stuff :)

I'll try the bleeding edge version right now and get back to you with an edit again :)

EDIT: indeed, that seems to be completely fixed in bleeding edge! Great to see!
The game still has some weird micro stutter issues but I don't think it's Proton's fault, a few months back there were some updates to the way frame limiting / refresh rate handles in Gamescope/SteamOS and I notice many games being affected by that (HL2 is an example too saldy), but this is an entirely other issue, as far as Proton goes, this looks like it's fixed, thank you so much!

If you feel like having the courage, you need to enter com_fixedtic 1 in the console or com_fixedtic=1 in the .ini, and then lock the gamecope limiter to 62 Hz to achieve 62 FPS, this will make the game depend on refresh rate speed, but since the original game was meant to run at 62.5 FPS, it's extremely close to the intended game speed :) It's a bit obscure to do this, but this way the game is very smooth and at (almost) intended speed, well, it was very smooth for the longest time until some updates dropped and maybe changed things idk.

**Comment by alasky17** [comment]
@Deus-nsf Thank you for the feedback and the testing!! I don't play this game regularly, so it is great to know that it seems completely fixed.  We might iterate on this fix a bit more as it was actually fixing a regression with a different game which still isn't 100% fixed ... if so, I'll let you know. If you notice any "weirdness" with the mouse control in bleeding-edge after playing more,  please let me know :)

**Comment by Deus-nsf** [comment]
For science I updated the Proton Exp build which is still configured with the bleeding edge branch, and the fix seems to hold so far :)
Now I have to really do something about that micro stutter issue (not doom 3 related), since I don't know what SteamOS version, it affects the compositor, no matter the refreshrate you choose, there's occasional, very visible stuttering, it's on every game I tried so far. Wasn't like this months ago (and for years before that change), I really owe myself to create an issue maybe on Gamescope's GitHub. It would be useful to enable tearing to see if it's a sync issue, but that feature has been broken for 2+ years at this point...
TLDR: Proton good, Gamescope/Game Mode eeeeh, not so much, at least I'd say it is the limiting factor on the Deck so far, at least in non VRR scenarios. (or maybe it's related to how input works... really hard to nail down this issue)

**Comment by Deus-nsf** [comment]
There was a major push that was made on the stable SteamOS branch recently, and I saw many occurences in the changelog mentionning the compositor and frame delivery, and how it would apparently fix things mostly for VRR.
I'm not sure what changes were made under the hood, but it seems to have completely fixed the presentation pacing issues the Deck had for a while (9 months+!) now! Now every game that can maintain the target refreshrate have perfect presentation and it's amazing, I'm so happy!

Now we only need the ability to restore the "allow tearing" functionality for latency enthusiasts!
